src/satisfiers/challenge.ts

```diff
@@ -0,0 +1,54 @@
+import { Satisfier } from 'lsat-js'
+import { decodeChallengeCaveat } from '../helpers'
+import { secp256k1 } from 'bcrypto'
+import zbase32 from 'zbase32'
+import { createHash } from 'crypto'
+const condition = 'challenge'
+
+// singleton to keep track of whether we are on the challenge caveat
+// or the answer caveat (the challenge caveat with the signature)
+let callCount = 0
+const sha256 = (msg: string | Buffer): Buffer =>
+  createHash('sha256')
+    .update(msg)
+    .digest()
+const MSG_PREFIX = 'Lightning Signed Message:'
+const challengeSatisfier: Satisfier = {
+  condition,
+  satisfyPrevious: (prev, curr) => {
+    const prevDecoded = decodeChallengeCaveat(prev.encode())
+    const currDecoded = decodeChallengeCaveat(curr.encode())
+    if (prevDecoded.challenge !== currDecoded.challenge) return false
+    if (prevDecoded.pubkey !== currDecoded.pubkey) return false
+    if (!currDecoded.signature) return false
+    return true
+  },
+  satisfyFinal: caveat => {
```

Comment:
@jacohend This method verifies the signature returned from the `signMessage` rpc, mimicking what the `verifyMessage` RPC would otherwise do.